#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "../protos/TrackOne.proto"

WorldInfo {
  info [
    "The MyBot robot, the robot used in the tutorial of Webots"
  ]
  title "MyBot"
}
Viewpoint {
  orientation -0.134139958323961 0.9561768519771188 0.26025429741691725 0.98453346273933
  position -1.0714293377202666 -0.768487963635642 2.521850201860629
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 0 0 1 5.307156870259904e-06
  floorSize 2 1
}
DEF GREEN_BOX Solid {
  translation 0.71 0.05 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 0.2189099510000001
        metalness 0
      }
      geometry DEF BOX0 Box {
        size 0.1 0.23 0.1
      }
    }
  ]
  name "green box"
  boundingObject Shape {
    appearance PBRAppearance {
      roughness 1.1102230246251565e-16
      metalness 0
    }
    geometry USE BOX0
  }
}
DEF BLUE_BOX Solid {
  translation -0.205929 -0.399983 0.05
  rotation 0 0 1 0.31
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 0.519341125
        metalness 0
      }
      geometry DEF BOX1 Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "blue box"
  boundingObject Shape {
    appearance PBRAppearance {
      roughness 1.1102230246251565e-16
      metalness 0
    }
    geometry USE BOX1
  }
}
DEF YELLOW_BOX Solid {
  translation -0.0231491 0.362202 0.05
  rotation 0 0 1 0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 0.03934117200000009
        metalness 0
      }
      geometry DEF BOX2 Box {
        size 0.3 0.05 0.1
      }
    }
  ]
  name "yellow box"
  boundingObject Shape {
    appearance PBRAppearance {
      roughness 1.1102230246251565e-16
      metalness 0
    }
    geometry USE BOX2
  }
}
DEF PINK_BOX Solid {
  translation 0.1 -0.34 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.1 0.3
        roughness 0.16827074099999995
        metalness 0
      }
      geometry DEF BOX3 Box {
        size 0.08 0.15 0.1
      }
    }
  ]
  name "red box"
  boundingObject Shape {
    appearance PBRAppearance {
      roughness 1.1102230246251565e-16
      metalness 0
    }
    geometry USE BOX3
  }
}
Robot {
  translation 0.2705117620267592 0.2032458479986225 -4.25807e-09
  rotation -2.82252e-08 3.50504e-10 1 -3.1399615529895324
  children [
    Camera {
      translation 0.0379144 0.002 0.0886947
      rotation 0 -1 0 -0.3927053061004253
      scale 0.217356 0.217356 0.217356
      children [
        Transform {
        }
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.1
            radius 0.05
          }
        }
      ]
      fieldOfView 0.785
      width 200
      height 200
    }
    Transform {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 738.567401352566
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          maxVelocity 11
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.04499999999999999 0.025
        rotation 0.7070874807914251 -0.7070876647481112 -0.007370811970096688 3.1268512422968366
        children [
          DEF WHEEL Transform {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 754.6959096713786
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          maxVelocity 11
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.045000000000000005 0.025
        rotation -0.3771099561272738 0.3771099273628031 -0.8459173622015965 1.7373545387894793
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
      }
    }
    DistanceSensor {
      translation 0.0491391 0.00150089 0.063
      rotation 7.1564999855133116e-06 -6.322439987201674e-05 0.9999999979757299 -0.024395307179585934
      children [
        DEF SENSOR Transform {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds2"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation -3.14419e-07 -7.19235e-09 1 -0.5007953071795859
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.029 -0.038 0.064799
      rotation 0.032427198091285726 0.04916889710584382 0.9982639412406639 -0.9360453071795867
      children [
        USE SENSOR
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.04 0.024 0.064
      rotation 0 0 1 0.523599
      children [
        USE SENSOR
      ]
      name "ds3"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.021 0.043 0.064
      rotation 0 0 -1 -1.047195307179586
      children [
        USE SENSOR
      ]
      name "ds4"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/samples/mybot/worlds/textures/mybot.png"
          ]
        }
        roughness 0.5
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.041 -0.015 0.038
            0.041 -0.015 0.023
            0.0455 0 0.023
            0.041 0.015 0.023
            0.041 0.015 0.038
            0.0455 0 0.038
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0.5 0
            1 0
            1 1
            0.5 1
            0 1
          ]
        }
        coordIndex [
          0, 1, 2, 5, -1, 5, 2, 3, 4, -1
        ]
        texCoordIndex [
          5, 0, 1, 4, -1, 4, 1, 2, 3, -1
        ]
        creaseAngle 0.9
      }
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  controller "keyboard_controller_collect"
}
TrackOne {
  translation 0 0.0129408 -0.007
  rotation 1 0 0 1.5708
}
